### PR TITLE
Improved docs content

### LINKED ISSUE #29: Migrate to BiomeJS
- [ ] Migrate our build systems to `@biomejs/biome`

### LINKED ISSUE #31: fix: correct all the API of the docs
Improve all the APIs of the components in the docs

### PR DESCRIPTION
**Pull request by LuckySharma014**
Closes #31 , Closes #29

- [x] Improved docs content
- [x] Improved props table

### CODE REVIEW

File: apps/docs/app/docs/components/field-control/page.md [hunk 1]
```diff
@@ -77,3 +77,25 @@ There are 3 `orientation` options in field control: `row` (default), `col` & `ro
...
+
+| Property    | Description                                   | Type                                  | Default |
+| ----------- | --------------------------------------------- | ------------------------------------- | ------- |
+| isDisabled  | Whether the component is disabled.            | `boolean`                             | `false` |
+| isLoading   | Whether the component is in a loading state.  | `boolean`                             | `false` |
+| isReadOnly  | Whether the component is in read-only mode.   | `boolean`                             | `false` |
+| isInvalid   | Whether the component is in an invalid state. | `boolean`                             | `false` |
+| isRequired  | Whether the component is required.            | `boolean`                             | `false` |
+| orientation | Orientation of the component.                 | `"col" `or `"row"` or `"row-reverse"` | `col`   |
+| name        | Name of the component.                        | `string`                              | -       |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/field-control/page.md:97]: Change description to - Name of the field

File: apps/docs/app/docs/components/hover-card/page.md [hunk 2]
```diff
@@ -34,3 +34,24 @@ import { HoverCard, HoverCardTrigger, HoverCardContent } from "@rafty/ui";
...
+This component is built on top of [Radix Hover Card Trigger](https://www.radix-ui.com/primitives/docs/components/hover-card#trigger)
+
+### HoverCardContent
+
+This component is built on top of [Radix Hover Card Content](https://www.radix-ui.com/primitives/docs/components/hover-card#content)
+
+| Property   | Description                          | Type                               | Default    |
+| ---------- | ------------------------------------ | ---------------------------------- | ---------- |
+| align      | Alignment of the component.          | `"center"` or `"start"` or `"end"` | `"center"` |
+| sideOffset | Horizontal offset for the component. | `number`                           | `4`        |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/hover-card/page.md:57]: Change description to - Side offset of the content from trigger

File: apps/docs/app/docs/components/input-field/page.md [hunk 3]
```diff
@@ -74,32 +74,43 @@ There are 3 `variant` options in input field: `solid`, `outline` (default) & `gh
...
-| Property   | Description                  | Type      | Default |
-| ---------- | ---------------------------- | --------- | ------- |
-| isUnstyled | Removes Style from component | `boolean` | false   |
+### InputField
+
+`InputField` composes the `input` component.
+
+| Property   | Description                                   | Type                                  | Default     |
+| ---------- | --------------------------------------------- | ------------------------------------- | ----------- |
+| size       | Size of the alert dialog trigger component.   | `"sm"` or `"md"` or `"lg"`            | `"md"`      |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/input-field/page.md:109]: Correct the description

File: apps/docs/app/docs/components/input-field/page.md [hunk 4]
```diff
@@ -74,32 +74,43 @@ There are 3 `variant` options in input field: `solid`, `outline` (default) & `gh
...
+| variant    | Style variant of the component.               | `"solid"` or `"outline"` or `"ghost"` | `"outline"` |
+| isDisabled | Whether the component is disabled.            | `boolean`                             | `false`     |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/input-field/page.md:111]: Change description to 'To manage disabled state"

File: apps/docs/app/docs/components/input-field/page.md [hunk 5]
```diff
@@ -74,32 +74,43 @@ There are 3 `variant` options in input field: `solid`, `outline` (default) & `gh
...
+| isRequired | Whether the component is required.            | `boolean`                             | `false`     |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/input-field/page.md:112]: Change description to 'To manage required state"

File: apps/docs/app/docs/components/input-field/page.md [hunk 6]
```diff
@@ -74,32 +74,43 @@ There are 3 `variant` options in input field: `solid`, `outline` (default) & `gh
...
+| isLoading  | Whether the component is in a loading state.  | `boolean`                             | `false`     |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/input-field/page.md:113]: Change description to 'To manage loading state"

File: apps/docs/app/docs/components/input-field/page.md [hunk 7]
```diff
@@ -74,32 +74,43 @@ There are 3 `variant` options in input field: `solid`, `outline` (default) & `gh
...
+| isReadOnly | Whether the component is in read-only mode.   | `boolean`                             | `false`     |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/input-field/page.md:114]: Change description to 'To manage read-only state"

File: apps/docs/app/docs/components/input-field/page.md [hunk 8]
```diff
@@ -74,32 +74,43 @@ There are 3 `variant` options in input field: `solid`, `outline` (default) & `gh
...
+| isUnstyled | Removes Style from component                  | `boolean`                             | `false`     |
+| isInvalid  | Whether the component is in an invalid state. | `boolean`                             | `false`     |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/input-field/page.md:116]: Change description to 'To manage invalid state"

File: apps/docs/app/docs/components/stat/page.md [hunk 9]
```diff
@@ -43,3 +43,36 @@ import { Stat, StatHelpText, StatIcon, StatLabel, StatValue } from "@rafty/ui";
...
+`StatHelpText` composes the `div` component.
+
+### StatIcon
+
+`StatIcon` composes the `svg` component.
+
+| Property | Description                             | Type     | Default |
+| -------- | --------------------------------------- | -------- | ------- |
+| height   | Height of the stat component in pixels. | `number` | `14`    |
+| width    | Width of the stat component in pixels.  | `number` | `14`    |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/stat/page.md:78]: Correct the description - "height/width of the icon"

File: apps/docs/app/docs/components/switch/page.md [hunk 10]
```diff
@@ -61,3 +61,19 @@ IsRequired prop is used to show select in required state.
...
+---
+
+### Switch
+
+This component is built on top of [Radix Switch](https://www.radix-ui.com/primitives/docs/components/switch#root)
+
+| Property       | Description                            | Type                       | Default |
+| -------------- | -------------------------------------- | -------------------------- | ------- |
+| size           | Size of the switch component.          | `"sm"` or `"md"` or `"lg"` | `"md"`  |
+| isDisabled     | Whether the component is disabled.     | `boolean`                  | `false` |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/switch/page.md:76]: Change description to "To manage disabled state"

File: apps/docs/app/docs/components/switch/page.md [hunk 11]
```diff
@@ -61,3 +61,19 @@ IsRequired prop is used to show select in required state.
...
+| defaultChecked | Initial state of the switch component. | `boolean`                  | `false` |
+| isRequired     | Whether the component is required.     | `boolean`                  | `false` |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/switch/page.md:78]: Change description to "To manage required state"

File: apps/docs/app/docs/components/switch/page.md [hunk 12]
```diff
@@ -61,3 +61,19 @@ IsRequired prop is used to show select in required state.
...
+| isReadOnly     | Whether the component is read-only.    | `boolean`                  | `false` |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/switch/page.md:79]: Change description to "To manage read-only state"

File: apps/docs/app/docs/components/switch/page.md [hunk 13]
```diff
@@ -61,3 +61,19 @@ IsRequired prop is used to show select in required state.
...
+
+### Switch
+
+This component is built on top of [Radix Switch](https://www.radix-ui.com/primitives/docs/components/switch#root)
+
+| Property       | Description                            | Type                       | Default |
+| -------------- | -------------------------------------- | -------------------------- | ------- |
+| size           | Size of the switch component.          | `"sm"` or `"md"` or `"lg"` | `"md"`  |
+| isDisabled     | Whether the component is disabled.     | `boolean`                  | `false` |
+| defaultChecked | Initial state of the switch component. | `boolean`                  | `false` |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/switch/page.md:77]: change description - "To manage default state of switch"

File: apps/docs/app/docs/components/tab/page.md [hunk 14]
```diff
@@ -74,30 +74,41 @@ Pass `isUnstyled` prop in a particular subcomponent to remove style from that su
...
-| ---------- | ------------------------------------------------------- | ------- | ------- |
-| isUnstyled | Removes style from whole component and its subcomponent | boolean | false   |
+This component is built on top of [Radix Tabs](https://www.radix-ui.com/primitives/docs/components/tabs#root)
 
-### List
+| Properties  | Description                                             | Type                       | Default      |
+| ----------- | ------------------------------------------------------- | -------------------------- | ------------ |
+| size        | Size of the tab component.                              | `"sm"` or `"md"` or `"lg"` | `"md"`       |
+| isUnstyled  | Removes style from whole component and its subcomponent | boolean                    | `false`      |
+| orientation | Orientation of the component.                           | `horizontal` or `vertical` | `horizontal` |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/tab/page.md:89]: - Add double quotes on type and default value
- Change description to - "Tab orientation"

File: apps/docs/app/docs/components/popover/page.md [hunk 15]
```diff
@@ -65,25 +65,47 @@ To make arrow on top of content visible you can use `showArrow` prop. It is a bo
...
+| isUnstyled  | Removes Style from component                 | `boolean`                                                | `false`       |
+| asChild     | Treats the component as a child element.     | `boolean`                                                | `false`       |
+
+### PopoverContent
+
+This component is built on top of [Radix Popover Content](https://www.radix-ui.com/primitives/docs/components/popover#content)
+
+| Property        | Description                                   | Type      | Default |
+| --------------- | --------------------------------------------- | --------- | ------- |
+| sideOffset      | Horizontal offset for the component.          | `number`  | `10`    |
```
[the remark this answers is not in the document]
Comment (Gajendra-singh-r) [reply apps/docs/app/docs/components/popover/page.md:104]: I've added "Side offset of the content from trigger" description

File: apps/docs/app/docs/components/tab/page.md [hunk 16]
```diff
@@ -74,30 +74,41 @@ Pass `isUnstyled` prop in a particular subcomponent to remove style from that su
...
+| variant     | Style variant of the component.                         | `line` or `enclosed`       | `line`       |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/tab/page.md:90]: - Add double quotes on type and default value
- Change description - "Tab style variant"

File: apps/docs/app/docs/components/table/page.md [hunk 17]
```diff
@@ -76,42 +76,68 @@ Pass `isUnstyled` prop in a particular subcomponent to remove style from that su
...
 
-| Property   | Description                  | Type     | Default |
-| ---------- | ---------------------------- | -------- | ------- |
-| isUnstyled | Removes Style from component | `string` | -       |
+`Table` composes the `table` component.
 
-### Body
+| Property   | Description                                              | Type                       | Default  |
+| ---------- | -------------------------------------------------------- | -------------------------- | -------- |
+| size       | Size of the switch component.                            | `"sm"` or `"md"` or `"lg"` | `"md"`   |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/table/page.md:93]: Correct the description

File: apps/docs/app/docs/components/table/page.md [hunk 18]
```diff
@@ -76,42 +76,68 @@ Pass `isUnstyled` prop in a particular subcomponent to remove style from that su
...
+| variant    | Style variant of the component.                          | `simple` or `striped`      | `simple` |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/table/page.md:94]: Add double quotes on type and default value

File: apps/docs/app/docs/components/tag-field/page.md [hunk 19]
```diff
@@ -29,3 +29,16 @@ import { TagField } from "@rafty/ui";
...
 
 {% example name="tag-field:usage" /%}
+
+## Props
+
+---
+
+### TagField
+
+`TagField` composes the `div` component.
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/tag-field/page.md:39]: Change here from div to rafty's Input

File: apps/docs/app/docs/components/textarea/page.md [hunk 20]
```diff
@@ -46,12 +46,21 @@ Pass `isDisabled` prop to put textarea in disabled state.
...
-| Property   | Description                  | Type      | Default |
-| ---------- | ---------------------------- | --------- | ------- |
-| isUnstyled | Removes style from component | `boolean` | `false` |
+`Textarea` composes the `textarea` component.
+
+| Property   | Description                                            | Type                                  | Default     |
+| ---------- | ------------------------------------------------------ | ------------------------------------- | ----------- |
+| size       | Size of the textarea component.                        | `"sm"` or `"md"` or `"lg"`            | `"md"`      |
+| variant    | Style variant of the component.                        | `"solid"` or `"outline"` or `"ghost"` | `"outline"` |
+| isDisabled | whether the component is disabled.                     | `boolean`                             | `false`     |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/textarea/page.md:61]: Change description to "To manage disabled state"

File: apps/docs/app/docs/components/textarea/page.md [hunk 21]
```diff
@@ -46,12 +46,21 @@ Pass `isDisabled` prop to put textarea in disabled state.
...
+| isLoading  | whether the component is in a loading state.           | `boolean`                             | `false`     |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/textarea/page.md:62]: Change description to "To manage loading state"

File: apps/docs/app/docs/components/textarea/page.md [hunk 22]
```diff
@@ -46,12 +46,21 @@ Pass `isDisabled` prop to put textarea in disabled state.
...
+| isRequired | Whether the textarea component is required.            | `boolean`                             | `false`     |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/textarea/page.md:63]: Change description to "To manage required state"

File: apps/docs/app/docs/components/textarea/page.md [hunk 23]
```diff
@@ -46,12 +46,21 @@ Pass `isDisabled` prop to put textarea in disabled state.
...
+| isReadOnly | Whether the textarea component is read-only.           | `boolean`                             | `false`     |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/textarea/page.md:64]: Change description to "To manage read-only state"

File: apps/docs/app/docs/components/textarea/page.md [hunk 24]
```diff
@@ -46,12 +46,21 @@ Pass `isDisabled` prop to put textarea in disabled state.
...
+| isInvalid  | Whether the textarea component is in an invalid state. | `boolean`                             | `false`     |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/textarea/page.md:65]: Change description to "To manage invalid state"

File: apps/docs/app/docs/components/toast/page.md [hunk 25]
```diff
@@ -59,3 +59,18 @@ There are 4 `severity` options in toast: `error`, `warning`, `info` & `success`.
...
+
+---
+
+### Toast
+
+`Toast` composes the `div` component.
+
+| Property | Description                    | Type                                                | Default |
+| -------- | ------------------------------ | --------------------------------------------------- | ------- |
+| severity | Severity level of the toast.   | `"error"` or `"success"` or `"warning"` or `"info"` | `false` |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/toast/page.md:73]: There is no default value of severity prop

File: apps/docs/app/docs/components/toggle-group/page.md [hunk 26]
```diff
@@ -62,17 +62,22 @@ Pass `isUnstyled` prop in a particular subcomponent to remove style from that su
...
 
-| Property   | Description                                              | Type    | Default |
-| ---------- | -------------------------------------------------------- | ------- | ------- |
-| isUnstyled | Removes style from whole component and its subcomponents | boolean | false   |
+This component is built on top of [Radix Toggle Group](https://www.radix-ui.com/primitives/docs/components/toggle-group#root)
 
-### Item
+| Property   | Description                                              | Type                       | Default |
+| ---------- | -------------------------------------------------------- | -------------------------- | ------- |
+| size       | Size of the tag component.                               | `"sm"` or `"md"` or `"lg"` | `"md"`  |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/toggle-group/page.md:75]: Correct the description

File: apps/docs/app/docs/components/table/page.md [hunk 27]
```diff
@@ -82,62 +82,62 @@ Pass `isUnstyled` prop in a particular subcomponent to remove style from that su
...
+`Table` composes the `<table>` component.
 
-| Property   | Description                                              | Type                       | Default  |
-| ---------- | -------------------------------------------------------- | -------------------------- | -------- |
-| size       | Size of the switch component.                            | `"sm"` or `"md"` or `"lg"` | `"md"`   |
-| variant    | Style variant of the component.                          | `simple` or `striped`      | `simple` |
-| isUnstyled | Removes style from whole component and its subcomponents | `boolean`                  | `false`  |
+| Property   | Description                                            | Type                       | Default  |
+| ---------- | ------------------------------------------------------ | -------------------------- | -------- |
+| size       | Size of the table component.                           | `"sm"` or `"md"` or `"lg"` | `"md"`   |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/table/page.md]: add to description - "and its subcomponents"

File: apps/docs/app/docs/components/stat/page.md [hunk 28]
```diff
@@ -72,7 +72,7 @@ The type prop is used to define type of stat, it has 3 options: `increase`, `dec
...
 `StatIcon` composes the `<svg>` component.
 
-| Property | Description                             | Type     | Default |
-| -------- | --------------------------------------- | -------- | ------- |
-| height   | Height of the stat component in pixels. | `number` | `14`    |
-| width    | Width of the stat component in pixels.  | `number` | `14`    |
+| Property | Description        | Type     | Default |
+| -------- | ------------------ | -------- | ------- |
+| height   | height of the icon | `number` | `14`    |
+| width    | width of the icon  | `number` | `14`    |
```
Comment (LuckySharma014) [inline apps/docs/app/docs/components/stat/page.md:78]: Capitalize first letters of height and width

**Pull request merged**